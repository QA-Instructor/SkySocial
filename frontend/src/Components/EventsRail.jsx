import React, { useRef } from 'react';
import '../resources/eventsRail.css';
import { MdArrowForwardIos, MdArrowBackIos } from "react-icons/md";
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types'
export const EventsRail = ({children,props}) => {
    let scrl = useRef(null);
    const [scrollX, setScrollX] = React.useState(0);
    const [scrollEnd, setScrollEnd] = React.useState(false);

    function slide(shift){
      scrl.current.scrollLeft += shift;
      setScrollX(scrollX + shift);
      if (Math.floor(scrl.current.scrollWidth - scrl.current.scrollLeft) <= scrl.current.offsetWidth) {
        setScrollEnd(true);
      } else {
        setScrollEnd(false);
      }
    };
  
    const checkScrollIsAtEnd = () => {
      setScrollX(scrl.current.scrollLeft);
      if (Math.floor(scrl.current.scrollWidth - scrl.current.scrollLeft) <= scrl.current.offsetWidth) {
        setScrollEnd(true);
      } else {
        setScrollEnd(false);
      }
    };
  
    return (
      <div fluid className='item-slider-container'>
        <div className='item-slider'>
          <div  onClick={() => slide(-230)} className={`left-arrow-left ${(scrollX < 1) ? 'is-disabled-hide' : ''}`}>
            <MdArrowBackIos size="70px" />
          </div>
          
          <div ref={scrl} onScroll={checkScrollIsAtEnd} className="item-container">
            {children}
          </div>
          <div className={`right-arrow-right ${(!scrollEnd) ? '' : 'is-disabled-hide'}`} onClick={() => slide(+230)}>
            <MdArrowForwardIos size="70px" />
          </div>
        </div>
      </div>
    );
};


export default EventsRail;
