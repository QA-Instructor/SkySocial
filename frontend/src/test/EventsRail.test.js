import React from 'react';
import { create } from 'react-test-renderer';
import EventsRail from '../Components/EventsRail';
afterEach(() => {
    // restore the spy created with spyOn
    jest.restoreAllMocks();
  });

describe('Event Tile Snapshot tests',()=>{
    test(`EventTile snapshot test`, () => {
        const testRenderer = create(<EventsRail>
            <div width="200px" height="200px">1</div>
            <div width="200px" height="200px">2</div>
            <div width="200px" height="200px">3</div>
            <div width="200px" height="200px">4</div>
            <div width="200px" height="200px">5</div>
            <div width="200px" height="200px">6</div>
            <div width="200px" height="200px">7</div>
            <div width="200px" height="200px">8</div>
            <div width="200px" height="200px">9</div>
        </EventsRail>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
});
