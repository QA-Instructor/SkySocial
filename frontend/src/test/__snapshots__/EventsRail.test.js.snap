// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Tile Snapshot tests EventTile snapshot test 1`] = `
<div
  className="item-slider-container"
  fluid={true}
>
  <div
    className="item-slider"
  >
    <div
      className="left-arrow-left is-disabled-hide"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="70px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="70px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </div>
    <div
      className="item-container"
      onScroll={[Function]}
    >
      <div
        height="200px"
        width="200px"
      >
        1
      </div>
      <div
        height="200px"
        width="200px"
      >
        2
      </div>
      <div
        height="200px"
        width="200px"
      >
        3
      </div>
      <div
        height="200px"
        width="200px"
      >
        4
      </div>
      <div
        height="200px"
        width="200px"
      >
        5
      </div>
      <div
        height="200px"
        width="200px"
      >
        6
      </div>
      <div
        height="200px"
        width="200px"
      >
        7
      </div>
      <div
        height="200px"
        width="200px"
      >
        8
      </div>
      <div
        height="200px"
        width="200px"
      >
        9
      </div>
    </div>
    <div
      className="right-arrow-right "
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="70px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="70px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
        />
      </svg>
    </div>
  </div>
</div>
`;
